src/Controller/Component/PrgComponent.php

```diff
@@ -2,26 +2,70 @@
 namespace Search\Controller\Component;
 
 use Cake\Controller\Component;
-use Cake\Event\Event;
 
 class PrgComponent extends Component
 {
+
+    /**
+     * Default config
```

Comment:
I'm not using `index` as default because of backward compatibility. If you don't do the check manually in the controller it will apply to all actions by default.